lib/iris/tests/unit/data_manager/test_DataManager.py

```diff
@@ -0,0 +1,602 @@
+        self.assertEqual(dm.dtype, np.dtype('bool'))
+        self.assertEqual(dm._lazy_array.dtype, np.dtype('int64'))
+
+
+class Test_ndim(tests.IrisTest):
+    def test_ndim_0(self):
+        real_array = np.array(0)
+        dm = DataManager(real_array)
+        self.assertEqual(dm.ndim, 0)
+        lazy_array = da.from_array(real_array, chunks=1)
+        dm = DataManager(lazy_array)
+        self.assertEqual(dm.ndim, 0)
+
+    def test_ndim_nd(self):
+        shape = (2, 3, 4)
+        real_array = np.arange(24).reshape(shape)
+        dm = DataManager(real_array)
+        self.assertEqual(dm.ndim, len(shape))
+        lazy_array = da.from_array(real_array, chunks=1)
+        dm = DataManager(lazy_array)
+        self.assertEqual(dm.ndim, len(shape))
+
+
+class Test_shape(tests.IrisTest):
+    def test_shape_scalar(self):
+        real_array = np.array(0)
+        dm = DataManager(real_array)
+        self.assertEqual(dm.shape, ())
+        lazy_array = da.from_array(real_array, chunks=1)
+        dm = DataManager(lazy_array)
+        self.assertEqual(dm.shape, ())
+
+    def test_shape_nd(self):
+        shape = (2, 3, 4)
+        real_array = np.arange(24).reshape(shape)
+        dm = DataManager(real_array)
+        self.assertEqual(dm.shape, shape)
+        lazy_array = da.from_array(real_array, chunks=1)
+        dm = DataManager(lazy_array)
+        self.assertEqual(dm.shape, shape)
+
+
+class Test__has_real_data_and_has_lazy_data(tests.IrisTest):
+    def setUp(self):
+        self.real_array = np.array(0)
+        self.lazy_array = da.from_array(self.real_array, chunks=1)
+
+    def test_with_lazy_array(self):
+        dm = DataManager(self.lazy_array)
+        self.assertFalse(dm.has_real_data())
+        self.assertTrue(dm.has_lazy_data())
+
+    def test_with_real_array(self):
+        dm = DataManager(self.real_array)
+        self.assertTrue(dm.has_real_data())
+        self.assertFalse(dm.has_lazy_data())
+
+
+class Test_lazy_data(tests.IrisTest):
+    def setUp(self):
+        self.real_array = np.array(0)
+        self.lazy_array = da.from_array(self.real_array, chunks=1)
+
+    def test_with_real_array(self):
+        dm = DataManager(self.real_array)
+        self.assertTrue(dm.has_real_data())
+        result = dm.lazy_data()
+        self.assertTrue(dm.has_real_data())
+        self.assertEqual(result, self.lazy_array)
+        self.assertTrue(dm.has_real_data())
+
+    def test_with_lazy_array(self):
+        dm = DataManager(self.lazy_array)
+        self.assertTrue(dm.has_lazy_data())
+        result = dm.lazy_data()
+        self.assertTrue(dm.has_lazy_data())
+        self.assertIs(result, dm._lazy_array)
+
+
+class Test_data__getter(tests.IrisTest):
+    def setUp(self):
+        shape = (2, 3, 4)
+        size = np.cumprod(shape)[-1]
+        self.real_array = np.arange(size).reshape(shape)
+        self.lazy_array = da.from_array(self.real_array, chunks=shape)
```

Comment:
Yeah, I hummed and hawed about this ... but was more focused on churning out as much test coverage as possible.

I think your right and I should rely on the goodness of `as_lazy_data` throughout